doc/on_chain_spec.md

```diff
@@ -669,3 +669,181 @@ that distributing their respective portions of the auction fee
+
+```haskell
+data BidTerms = BidTerms
+  { bidderPKH :: PubKeyHash
+  -- ^ PubKeyHash of whoever submitted the bid
+  , bidderVK :: BuiltinByteString
+  -- ^ New field. Verification Key of whoever submitted the bid
+  , bidPrice :: Natural
+  -- ^ Which price did the bidder set to buy the auction lot?
+  , bidderSignature :: BuiltinByteString
+  -- ^ New field. Represents the signed payload by the bidder
+  , sellerSignature :: BuiltinByteString
+  -- ^ Represents the signed payload by the seller
+  }
+```
+
+This signature can be generated by the bidder,
+by signing with their signing key,
+a tuple consisting of the `hydraHeadId`, `bidderPKH` and `bidAmount`
+
+```haskell
+generateBidderSignature :: SigningKey PaymentKey -> CurrencySymbol -> Natural -> PubKeyHash -> BuiltinByteString
+generateBidderSignature skey = toBuiltin . (dsign skey) . bidderSignatureMessage
+
+bidderSignatureMessage :: CurrencySymbol -> Natural -> PubKeyHash -> BuiltinByteString
+bidderSignatureMessage voucherCS bidAmount bidderPKH = toByteString voucherCS <> toByteString bidAmount <> toBysteString bidderPKH
+```
+
+We also need to extend the previous implementation of `sellerSignatureMessage` to also include the bidder verification key in the signature.
+We will use this to verify the association between bidder PKH and VK on-chain.
+
+```haskell
+generateSellerSignature :: SigningKey PaymentKey -> CurrencySymbol -> PubKeyHash -> SigningKey VerificationKey ->  BuiltinByteString
+generateSellerSignature skey = toBuiltin . (dsign skey) . sellerSignatureMessage
+
+sellerSignatureMessage :: CurrencySymbol -> PubKeyHash -> SigningKey VerificationKey -> BuiltinByteString
+sellerSignatureMessage voucherCS bidderPKH bidderVK = toByteString voucherCS <> toByteString bidderVK <> toBysteString bidderPKH
+```
+
+We now extend the Standing Bid Validator, to use on-chain verification
+to check that each new standing bid datum, is signed by the correct bidder.
+
+In particular, we can change `validNewBid` in the following way:
+
+```haskell
+validNewBid
+  :: AuctionTerms
+  -> CurrencySymbol
+  -> StandingBidState
+  -> StandingBidState
+  -> Bool
+validNewBid AuctionTerms{..} voucherCS oldStandingBidState newStandingBidState =
+  isJust $ do
+    let StandingBidState oldApprovedBidders oldBid = oldStandingBidState
+    let StandingBidState newApprovedBidders newBid = newStandingBidState
+    BidTerms bidderPKH bidderVK newPrice bidderSignature sellerSignature <- newBid
+    -- The seller has allowed the bidder to participate in the auction
+    guard $ verifyEd25519Signature (sellerVK terms) (sellerSignatureMessage voucherCS bidderPKH bidderVK) sellerSignature
+    -- The bidder has correctly signed the datum
+    guard $ verifyEd25519Signature bidderVK (bidderSignatureMessage voucherCS newPrice bidderPKH) bidderSignature
+    -- The new bid respects the starting price and minimum bid increment
+    -- conditions in the auction terms.
+    guard $ case oldBid of
```

Comment:
I plan to do same with `validAuctionTerms` for off-chain usage.

Reply:
I propose we do it when we need it.... we probably will at some point, but there is no reason to do it earlier. It should be easy to separate this into a function if needed

Reply:
> we probably will at some point

It is already used at `staging`. And I always know that it will be used, because parsing tx errors for showing UX is harder.